kubernetes/kubetest/mock.go

```diff
@@ -279,9 +279,19 @@ func FakeIstioAmbientAnnotations() map[string]string {
 }
 
 func FakeNamespace(name string) *core_v1.Namespace {
+	return FakeNamespaceWithLabels(name, nil)
+}
+
+func FakeNamespaceWithLabels(name string, labels map[string]string) *core_v1.Namespace {
```

Comment:
nit: you don't need to remove this but this can often end up with having many different methods for setting all the fields in a namespace e.g. `FakeNamespace(name string, labels, annotations, hasInjection, revisionLabel... etc.)` where it's simple enough to just do this where it's needed:
```
ns := FakeNamespace("")
ns.Labels = map[string]string{...}
ns.Annotations
```
or define the whole struct
```
corev1.Namespace{{ObjectMeta: ...}}
```

Reply:
I added this specifically to make things still one line of code AND avoid devs having to remember to always add the metadata.name label (which all k8s namespaces have and which our discovery selector stuff utilize - I suspect devs writing code in the future won't even know that sometimes namespaces need to have that label so they won't add it).

The issue here is we had a bunch of corev1.Namespaces all over the place, and we always want this `kubernetes.io/metadata.name` label in there - yes, I could have added that label in there without a new Fake function, but I didn't want people to have to remember to do that every time in the future. So FakeNamespace() is the way to do it..

BUT! Then you have tests that actually need additional labels. But we don't want to (a) lose the metadata.name and (b) require the dev to remember to add it. Hence, FakeNamespaceWithLabels.

So this won't grow with additional parameters - because the only reason "WithLabels" exists is because we want to support adding custom labels but NOT override the default label we always want in there. Because we don't have, say, default annotations or default whatever in our fake namespaces, the developer is free to do what you say with the returned namespace to add that stuff and no need to add parameters to the function.

ns := FakeNamespaceWithLabels(...labels in addition to the metadata.name that I get for free...)
ns.Annotations ...
ns...whatever

So, again, this is only to help the devs always get that "metadata.name" label in namespaces that are created - the WithLabels is merely to avoid the case where the dev needs a custom label, too. And you can do this with one line of code with the new functions.